File: API/Entities/OrderAggregate/Order.cs
namespace API.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public string BuyerId { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public ShippingAddress ShipToAddress { get; set; }
    public int DeliveryFee { get; set; }
    public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public decimal Subtotal { get; set; }
    public decimal GetTotal()
    {
        return Subtotal + DeliveryFee;
    }
}

File: client-app/src/app/models/order.ts
export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zipCode: string;
    country: string;
    phoneNumber: string;
}

export interface OrderItem {
    productId: number;
    productName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export interface Order {
    id: number;
    buyerId: string;
    orderDate: string;
    shipToAddress: ShippingAddress;
    deliveryFee: number;
    orderItems: OrderItem[];
    status: string;
    subtotal: number;
    total: number;
}

